### ISSUE TITLE
Executing JavaScript files locally

### ISSUE THREAD

**Issue by ghost** [e-conomic/client-side-application-course#41]
If you are using Chrome and try to put your JavaScript code in a separate .js file, this will fail when you run a local html file including that script. It will result in this error:

XMLHttpRequest cannot load file:///C:/Git/e-conomic/client-side-application-course/BennyTordrupVisma/Week2/Ass3/script.js. Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https, chrome-extension-resource.

This is a security feature in Chrome that it does not allow cross-origin requests.

However, this can be circumvented - also without hosting a local web server as described in [this thread](http://stackoverflow.com/questions/3102819/disable-same-origin-policy-in-chrome).

Basically, the trick is to create a shortcut for Chrome where you specify the following parameters:

--user-data-dir="C:/Chrome dev session" --disable-web-security

You have to have the directory "C:\Chrome dev session" created on disk.

What it does is that it launches Chrome in a new session with user data stored in the specified directory. the second parameter turns off web security (and hence allows cross-origin requests).

BEWARE - THIS SHOULD ONLY BE USED TO TEST SCRIPTS LOCALLY.

Do not use this instance of the browser to go on the internet. Your computere will be volnurable.

**Comment by hulvei3** [comment]
Great - thanks for sharing!

Seems a bit risky though as Chrome usually does all kinds of syncing. I don't think you can prevent Chrome to work _entirely_ offline. But definitely useful for this usecase

**Comment by ghost** [comment]
As long as you do not log in to Chrome, I think the syncing is minimal.And specifying the --user-data-dir will separate it from your normal Chrome instance.
